### PR TITLE
Refactored and restructured a bit to make app.ts less cluttered

### PR DESCRIPTION
**Pull request by skeltonath**
Addresses these issues: #6 #7

### CODE REVIEW

File: src/api/server.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import express from 'express';
+import { RifmDAO } from '../data/rifmDAO';
+import { SpotifyDAO } from '../data/spotifyDAO';
+
+export class APIServer {
+    private port: string;
+    private spotify: SpotifyDAO;
+    private rifm: RifmDAO;
+
+    constructor(port: string, spotify: SpotifyDAO, rifm: RifmDAO) {
```
Comment (nfadili) [inline src/api/server.ts]: Fun optional ts tidbit: You can declare all of this like:
```
constructor(private spotify: SpotifyDAO, private rifm: RifmDAO, private port: string = '3000') {}
```

Comment (nfadili) [reply src/api/server.ts]: And not need the members declared above

File: src/bot/bot.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    private sendAlbumEmbed(message: Discord.Message, i: number, albumList: SpotifyAlbumInfo[]) {
+        const filter = (reaction: Discord.MessageReaction, user: Discord.User) => user.id !== message.author.id &&
+            [ARROW_LEFT, ARROW_RIGHT].includes(reaction.emoji.name);
+        const album = albumList[i];
+        const albumEmbed = new Discord.RichEmbed()
+            .setColor('#1DB954')
+            .setTitle(album.name)
+            .setURL(album.url)
+            // .setDescription(album.artist)
```
Comment (nfadili) [inline src/bot/bot.ts]: Meant to be commented out?

File: src/api/server.ts (hunk 1 above)
Comment (skeltonath) [reply src/api/server.ts]: 🤯

File: src/data/spotifyDAO.ts [hunk 3]
```diff
@@ -35,21 +38,21 @@ export class SpotifyAPI {
...
-                    resolve(albums);
-                }, (err) => {
-                    console.error(err);
-                    reject(err);
+            .then((data) => {
+                const albums = data.body.albums.items.map((item) => {
+                    return {
+                        id: item.id,
+                        name: item.name,
+                        // artist: item.artists[0].name,
```
Comment (nfadili) [inline src/data/spotifyDAO.ts]: Meant to be commented out?

File: src/data/spotifyDAO.ts [hunk 4]
```diff
@@ -20,8 +20,11 @@ export class SpotifyAPI {
         // Retrieve an access token.
         this.api.clientCredentialsGrant().then(
             (data) => {
-                console.log('The access token expires in ' + data.body.expires_in);
-                console.log('The access token is ' + data.body.access_token);
+                console.log('Spotify access token expires in ' + data.body.expires_in);
+                console.log('Spotify access token is ' + data.body.access_token);
+                setTimeout(() => {
+                    console.log('Spotify access token expired!');
+                }, data.body.expires_in * 1000);
```
Comment (nfadili) [inline src/data/spotifyDAO.ts]: Was gonna ask if you had an issue for this but you do :)

File: src/data/spotifyDAO.ts (hunk 3 above)
Comment (skeltonath) [reply src/data/spotifyDAO.ts]: yup. That type is broken and needs to be fixed. The object returned by Spotify has the `artist` field, but the type doesn't define it, so it won't build.

File: src/bot/bot.ts (hunk 2 above)
Comment (skeltonath) [reply src/bot/bot.ts]: See comment below.

File: src/data/spotifyDAO.ts (hunk 4 above)
Comment (skeltonath) [reply src/data/spotifyDAO.ts]: Yeah haha, didn't wanna fix it here yet.

**Pull request merged**